mysql-test/suite/innodb/t/undo_leak_fail.test

```diff
@@ -0,0 +1,96 @@
+--source include/have_innodb.inc
+--source include/have_sequence.inc
+--source include/have_debug.inc
+--source include/not_embedded.inc
+
+call mtr.add_suppression("InnoDB: Cannot free the unused segments in system tablespace because a previous shutdown was not with innodb_fast_shutdown=0");
+call mtr.add_suppression("InnoDB: Cannot free the unused segment in system tablespace due to Data structure corruption");
+call mtr.add_suppression("InnoDB: Failed to free the unused segment");
+
+let $restart_parameters=--debug_dbug=d,undo_segment_leak;
+--source include/restart_mysqld.inc
+SET GLOBAL INNODB_FILE_PER_TABLE=0;
+CREATE TABLE t1(f1 INT NOT NULL, f2 INT NOT NULL)STATS_PERSISTENT=0, ENGINE=InnoDB;
+INSERT INTO t1 SELECT seq, seq FROM seq_1_to_4096;
+INSERT INTO t1 SELECT seq, seq FROM seq_1_to_4096;
+INSERT INTO t1 SELECT seq, seq FROM seq_1_to_4096;
+INSERT INTO t1 SELECT seq, seq FROM seq_1_to_4096;
+INSERT INTO t1 SELECT seq, seq FROM seq_1_to_4096;
+INSERT INTO t1 SELECT seq, seq FROM seq_1_to_4096;
+UPDATE t1 SET f1 = f1 + 1 WHERE f1 > 1000;
+UPDATE t1 SET f2 = f2 + 1 WHERE f1 > 1000;
+UPDATE t1 SET f1 = f2 + 1 WHERE f1 > 1000;
+UPDATE t1 SET f2 = f1 + 1 WHERE f1 > 1000;
+DELETE FROM t1;
+DROP TABLE t1;
+SELECT NAME, FILE_SIZE FROM information_schema.innodb_sys_tablespaces WHERE SPACE = 0;
+let $shutdown_timeout=0;
+let $restart_parameters=--debug_dbug=d,unused_undo_free_fail_1;
+--source include/restart_mysqld.inc
+
+let $shutdown_timeout=;
+--echo # Fail to free the segment due to previous shutdown
+--let SEARCH_PATTERN= \[Warning\] InnoDB: Cannot free the unused segments in system tablespace because a previous shutdown was not with innodb_fast_shutdown=0
```

Comment:
The `\` in this and other `SEARCH_PATTERN` assignments had better be doubled. Yes, currently `\` before a character that does not require escaping in a string literal may be interpreted as a literal `\`, but I think that it is bad practice to rely on that. While this is Perl and `mtr` instead of `grep` and `/bin/sh´, I think that it is useful to refer to https://superuser.com/questions/1063992/backslash-in-single-quotes-giving-unpredictable-output-in-shell-script-linux